azure-pipelines.yml

```diff
@@ -0,0 +1,52 @@
+trigger:
+  - master
+
+jobs:
+- job: Build
+  timeoutInMinutes: 360
+  strategy:
+    matrix:
+      windows:
+        imageName: 'vs2017-win2016'
+      mac:
+        imageName: 'macos-10.14'
+      linux:
+        imageName: 'ubuntu-16.04'
+
+  pool:
+    vmImage: $(imageName)
+
+  steps:
+    - script: |
+        powershell -Command "$ProgressPreference = 'SilentlyContinue'; iwr -outf %TEMP%\LLVM-8.0.0-win64.exe https://rust-lang-ci2.s3.amazonaws.com/rust-ci-mirror/LLVM-8.0.0-win64.exe"
+        set CLANG_DIR=%CD%\citools\clang-rust
+        start "" /WAIT %TEMP%\LLVM-8.0.0-win64.exe /S /NCRC /D=%CLANG_DIR%
+        echo ##vso[task.prependpath]%CLANG_DIR%/bin
+      condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'))
+      displayName: Install clang (Windows)
+    - bash: |
+        set -e
+        curl -f http://releases.llvm.org/8.0.0/clang+llvm-8.0.0-x86_64-apple-darwin.tar.xz | tar xJf -
+        export CLANG_DIR=`pwd`/clang+llvm-8.0.0-x86_64-apple-darwin/bin
+        echo "##vso[task.prependpath]$CLANG_DIR"
+      displayName: Install clang (OSX)
+      condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+    - bash: |
+        set -e
+        curl -f http://releases.llvm.org/8.0.0/clang+llvm-8.0.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz | tar xJf -
+        export CLANG_DIR=`pwd`/clang+llvm-8.0.0-x86_64-linux-gnu-ubuntu-16.04/bin
+        echo "##vso[task.prependpath]$CLANG_DIR"
+      displayName: Install clang (Linux)
+      condition: and(succeeded(), eq(variables['Agent.OS'], 'Linux'))
+    - bash: |
+        set -e
+        rustup update stable
+        rustup default stable
+        rustup component add llvm-tools-preview
```

Comment:
I don't understand why rust is has anything to do with installing `llvm-nm` . Are there not official llvm releases we can use?

Reply:
Yeah, this certainly requires some explanation :)

The reason this is needed is that, weirdly, the LLVM release for Windows doesn't include `llvm-nm`. Rust's `llvm-tools-preview` component includes `llvm-nm`, and Azure Pipelines machines have Rust preinstalled, so was the easiest way to make `llvm-nm` available that I could come up with.

I'll add a comment explaining this.